### PR TITLE
Enhancement | ECS EFS Attack Scenario

### PR DESCRIPTION
**Pull request by andrew-aiken**
#### Changes
- EFS can only be mounted from the subnet with the "admin" EC2 instance.
  - The Ruse EC2 instance we moved to another subnet & EFS was whitelisted for the admin subnet.
  - https://github.com/RhinoSecurityLabs/cloudgoat/issues/107
- Formatting & Cleanup
  - Linting
  - Move to jsonencode()

#### Testing
Terraform `1.5.6`

### CODE REVIEW

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 1]
```diff
@@ -3,127 +3,157 @@
 
 resource "aws_iam_role" "cg-ec2-ruse-role" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:4]: jsonencode, formatting, & combine policy attachments

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 2]
```diff
@@ -3,127 +3,157 @@
...
 }
 
+resource "aws_iam_instance_profile" "cg-ec2-ruse-instance-profile" {
+  name = "cg-ecsTaskExecutionRole-instance-profile-${var.cgid}"
+  role = aws_iam_role.cg-ec2-ruse-role.name
+}
 
 #IAM Admin Role
-
 resource "aws_iam_role" "cg-efs-admin-role" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:38]: jsonencode, formatting, & combine policy attachments

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 3]
```diff
@@ -3,127 +3,157 @@
...
-      Stack = "${var.stack-name}"
-      Scenario = "${var.scenario-name}"
-  }
+
+  tags = merge(local.default_tags, {
+    Name = "cg-ec2-role-${var.cgid}"
+  })
 }
 
+resource "aws_iam_instance_profile" "cg-ec2-ruse-instance-profile" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:32]: Move from ec2 file

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 4]
```diff
@@ -3,127 +3,157 @@
...
   ]
 }
-EOF
-  tags = {
-      Name = "cg-ec2-efsUser-role-${var.cgid}"
-      Stack = "${var.stack-name}"
-      Scenario = "${var.scenario-name}"
-  }
+
+resource "aws_iam_instance_profile" "cg-efs-admin-instance-profile" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:65]: Move from ec2 file

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 5]
```diff
@@ -3,127 +3,157 @@
...
+  name = "cg-efs-admin-instance-profile-${var.cgid}"
+  role = aws_iam_role.cg-efs-admin-role.name
 }
 
 ###### IAM Lambda Role ######
-
 resource "aws_iam_role" "cg-lambda-role" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:71]: jsonencode, formatting, & combine policy attachments

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 6]
```diff
@@ -3,127 +3,157 @@
...
-}
-EOF
+
+  tags = merge(local.default_tags, {
+    Name = "cg-lambda-role-${var.cgid}"
+  })
 }
 
 #Iam Role Policy for ec2 "ruse-box"
 resource "aws_iam_policy" "cg-ec2-ruse-role-policy" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:102]: jsonencode & formatting

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 7]
```diff
@@ -3,127 +3,157 @@
...
+      ]
+    }
+  )
+
+  tags = merge(local.default_tags, {
+    Name = "cg-ec2-ruse-role-policy-${var.cgid}"
+  })
 }
 
 resource "aws_iam_policy" "cg-efs-admin-role-policy" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:137]: jsonencode, formatting, & tags

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 8]
```diff
@@ -134,68 +164,69 @@ POLICY
 #IAM Role
 resource "aws_iam_role" "cg-ecs-role" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:165]: jsonencode & tags

File: scenarios/ecs_efs_attack/terraform/iam.tf [hunk 9]
```diff
@@ -134,68 +164,69 @@ POLICY
...
-  }
+  )
+
+  tags = merge(local.default_tags, {
+    Name = "cg-ecs-role-${var.cgid}"
+  })
 }
+
 #Iam Role Policy
 resource "aws_iam_policy" "cg-ecs-role-policy" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/iam.tf:189]: jsonencode & tags

File: scenarios/ecs_efs_attack/terraform/lambda.tf [hunk 10]
```diff
@@ -1,50 +1,35 @@
-
-
-resource "aws_iam_role_policy_attachment" "cg-lambda-policy" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/lambda.tf:3]: Move to iam file

File: scenarios/ecs_efs_attack/terraform/lambda.tf [hunk 11]
```diff
@@ -1,50 +1,35 @@
...
-}
-
-resource "aws_iam_role_policy_attachment" "cg-lambda-policy3" {
-  role       = "${aws_iam_role.cg-lambda-role.name}"
-  policy_arn = "arn:aws:iam::aws:policy/AmazonElasticFileSystemClientReadWriteAccess"
-}
-
-
 # A lambda function connected to an EFS file system
 resource "aws_lambda_function" "efs_upload" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/lambda.tf:2]: Formatting

File: scenarios/ecs_efs_attack/terraform/lambda.tf [hunk 12]
```diff
@@ -54,35 +39,24 @@ resource "aws_cloudwatch_event_rule" "cg_insert_file_every_three_minutes" {
...
   statement_id  = "AllowExecutionFromCloudWatch"
   action        = "lambda:InvokeFunction"
-  function_name = "${aws_lambda_function.efs_upload.function_name}"
+  function_name = aws_lambda_function.efs_upload.function_name
   principal     = "events.amazonaws.com"
-  source_arn    = "${aws_cloudwatch_event_rule.cg_insert_file_every_three_minutes.arn}"
+  source_arn    = aws_cloudwatch_event_rule.cg_insert_file_every_three_minutes.arn
 }
-
-# # Invoke the lambda function
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/lambda.tf:74]: Remove comment

File: scenarios/ecs_efs_attack/terraform/outputs.tf [hunk 13]
```diff
@@ -1,11 +1,10 @@
-
 #Required: Always output the AWS Account ID
 output "cloudgoat_output_aws_account_id" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/outputs.tf:2]: Formatting

File: scenarios/ecs_efs_attack/terraform/provider.tf [hunk 14]
```diff
@@ -1,4 +1,4 @@
 provider "aws" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/provider.tf:1]: formatting

File: scenarios/ecs_efs_attack/terraform/variables.tf [hunk 15]
```diff
@@ -1,32 +1,44 @@
-#Required: AWS Profile
 variable "profile" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/variables.tf:1]: Adding type & descriptions

File: scenarios/ecs_efs_attack/terraform/vpc.tf [hunk 16]
```diff
@@ -1,61 +1,60 @@
 resource "aws_vpc" "cg-vpc" {
```
Comment (andrew-aiken) [inline scenarios/ecs_efs_attack/terraform/vpc.tf:1]: Formatting and tagging

**Review by andrew-aiken — COMMENTED** [review]
Added generic comments of changes. @ me if there are any questions

**Review by jdearmas — APPROVED** [review]

**Pull request merged**